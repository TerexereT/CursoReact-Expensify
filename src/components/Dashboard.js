import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDBPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDBPage