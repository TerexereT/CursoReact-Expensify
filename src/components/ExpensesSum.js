import React from 'react'
import numeral from 'numeral'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const aux = expenseCount === 1 ? 'expensa' : 'expensas'
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viendo <span>{expenseCount}</span> {aux} de un total de <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button-layout" to="/create">Agregar expensa</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect (mapStateToProps)(ExpensesSummary)
