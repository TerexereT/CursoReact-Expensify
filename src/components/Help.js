import React from 'react'

const HelpExpenseDBPage = () => (
    <div>
        esto viene de Help Dashboard
    </div>
)

export default HelpExpenseDBPage