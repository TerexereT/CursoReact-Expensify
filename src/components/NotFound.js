import React from 'react'
import {Link} from 'react-router-dom'

const NotFoundDBPage = () => (
    <div>
        error 404! Page Not Found
        - <Link to="/">Home</Link>
    </div>
)

export default NotFoundDBPage