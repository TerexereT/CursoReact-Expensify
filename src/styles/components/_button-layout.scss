.button-layout {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-l;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}


.button--link {
    background: none;
    
}

.button-layout--secondary {
    @extend .button-layout;
    background: rgb(201, 0, 0);
}