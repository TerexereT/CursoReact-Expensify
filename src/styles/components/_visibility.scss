.show-for-mobile {
    @media (min-width: $desktopBp) {
        display: none;
    }
}

.show-for-desktop {
    @media (max-width: $desktopBp - .01rem){
        display: none;
    }
}