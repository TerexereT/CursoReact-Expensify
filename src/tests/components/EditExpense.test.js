import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import {EditExpenseDBPage} from '../../components/EditExpense'

let editExpense, removeExpense, history, wrapper

beforeEach(() => {
    editExpense=jest.fn(),
    removeExpense=jest.fn(),
    history={push: jest.fn()},
    wrapper =  shallow(
        <EditExpenseDBPage
        editExpense={editExpense}
        removeExpense={removeExpense}
        history={history}
        expense={expenses[2]}
        />
    )
})

test('render editexpense', () =>{
    expect(wrapper).toMatchSnapshot()
})

test('handle editexpense', () =>{
   wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2])
   expect(history.push).toHaveBeenLastCalledWith('/') 
   expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2])
})

test('handle removeexpense', () =>{
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/') 
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[2].id
    })
})