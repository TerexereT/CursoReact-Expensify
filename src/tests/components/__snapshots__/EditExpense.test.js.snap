// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render editexpense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 3321,
        "createdAt": 345600000,
        "description": "estas",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Eliminar
  </button>
</div>
`;
